#! /usr/bin/env python
PACKAGE='ece6460_ekf_example'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#        Name         Type      Level  Description                Default  Min    Max
gen.add("q_pos",      double_t, 0,    "Process noise covariance", 10.0,    0.0,   100.0)
gen.add("q_heading",  double_t, 0,    "Process noise covariance", 1.0,     0.0,   100.0)
gen.add("q_speed",    double_t, 0,    "Process noise covariance", 0.1,     0.0,   100.0)
gen.add("q_yaw_rate", double_t, 0,    "Process noise covariance", 10.0,    0.0,   100.0)
gen.add("r_gps",      double_t, 0,    "Measurement covariance",   5.0,     0.0,   100.0)
gen.add("r_speed",    double_t, 0,    "Measurement covariance",   0.01,    0.0,   100.0)
gen.add("r_yaw_rate", double_t, 0,    "Measurement covariance",   0.01,    0.0,   100.0)

exit(gen.generate(PACKAGE, PACKAGE, "EkfExample"))
